drivers/SmartThings/harman-luxury/src/init.lua

```diff
@@ -210,15 +200,25 @@ local function check_for_updates(device)
   end
 end
 
-local function create_check_for_updates_thread(device)
-  -- stop old timer if one exists
-  stop_check_for_updates_thread(device)
+local function check_for_updates_manager(driver)
+  -- get all available drivers
+  local devices = driver:get_devices()
 
-  log.info(string.format("create_check_for_updates_thread: dni=%s", device.device_network_id))
-  local new_timer = device.thread:call_on_schedule(const.UPDATE_INTERVAL, function()
-    check_for_updates(device)
-  end, "value_updates_timer")
-  device:set_field(const.UPDATE_TIMER, new_timer)
+  -- loop all devices and update them using polling
+  if next(devices) ~= nil then
+    for _, device in pairs(devices) do
+      local state = device:get_field(const.STATUS)
+      local device_dni = device.device_network_id
+      if state then
+        log.debug(string.format("polling values update for \"%s\"", device_dni))
+        check_for_updates(device)
```

Comment:
If one device is non-responsive, it will prevent updates of other devices from happening until the timeout has occurred. This is a problem with having a single "task" to handle all device updates, and IMO is a really good reason to have a single task per device to handle updates.